pkg/server/pgurl/pgurl_test.go

```diff
@@ -64,6 +64,49 @@ func TestURL(t *testing.T) {
 	})
 }
 
+func TestPassword(t *testing.T) {
+	u := New()
+
+	enabled, _, _ := u.GetAuthnPassword()
+	require.False(t, enabled)
+
+	u.WithAuthn(AuthnPassword(true, "abc"))
+
+	enabled, hasp, p := u.GetAuthnPassword()
+	require.True(t, enabled)
+	require.True(t, hasp)
+	require.Equal(t, p, "abc")
+
+	u.ClearPassword()
+	enabled, hasp, p = u.GetAuthnPassword()
+	require.True(t, enabled)
+	require.False(t, hasp)
```

Comment:
does this need a `require.Equal(t, p, "")` ?

Reply:
if "haspassword" is false, there's no API guarantee about the remaining value in the password field itself.